src/js/controller/controller.js

```diff
@@ -489,7 +493,8 @@ Object.assign(Controller.prototype, {
 
             const adState = _getAdState();
             const pauseReason = _model.get('pauseReason');
-            if (adState && pauseReason === 'viewable' && playReason !== 'interaction') {
+            const autoPauseAds = _model.get('autoPause').pauseAds;
+            if (adState && !autoPauseAds && pauseReason === 'viewable' && playReason !== 'interaction') {
```

Comment:
Why are you adding `pauseAds` logic in the `play` command? I would expect this logic to be outside of the `play` method where necessary. Doing it here will block API calls from toggling ad playback when pauseAds is false which we don't want.

Even the logic before this change seems really sketchy and should be scoped to (inside) the next `if(adState)` block.

Reply:
I didn't consider this `pauseAds` 'logic' because this isn't pausing anything - it's more the logic for resuming playback if `pauseAds` is set. However, if you feel like it would be better if I abstract this out, then I will do so.

Reply:
> Doing it here will block API calls from toggling ad playback when pauseAds is false which we don't want

Reply:
If `pauseAds` is true and I pause an ad by scrolling out of view, I'm able to call `.play()` in the console and have it resume.

Reply:
@robwalch - I just tested a bunch with `pauseAds: false` and everything worked as expected.

Reply:
It's seems like the `playReason` should be `"viewable"` too here. No?

What if you undid the changes to this if statement, and instead did not call play() on viewable when `!autoPauseAds`? (`_checkPlayOnViewable` -> `if (viewable) {`)

Reply:
Are you talking about in this conditional?

`else if (!adState || model.get('autoPause').pauseAds)`

If so, then I'm not quite understanding because we would still need to call `play()` if `pauseAds: false` if we're not in ad playback.